addons/spreadsheet/static/src/helpers/model.js

```diff
@@ -69,6 +69,11 @@ export async function freezeOdooData(model) {
             const sheetId = sheet.id;
             const position = { sheetId, col, row };
             const evaluatedCell = model.getters.getEvaluatedCell(position);
+            if (evaluatedCell.type === "text" && !isNaN(evaluatedCell.value)) {
```

Comment:
I think we can use `isNumber`.
It's more readable (double negation with `!` and *is**N**(ot)aN*).
Also, does this need to account for the locale ? I'm not sure how are hard-coded strings are imported.
```suggestion
            if (evaluatedCell.type === "text" && isNumber(evaluatedCell.value)) {
```

Reply:
isNumber is not exported. I didn't want to make an export PR (2+ weeks to have everything fixed) just for some niche case that only occurs in 17.0. I'll check if it breaks with dates as well tough i'm 99% sure it's also the case